.github/workflows/publish-package.yml

```diff
@@ -0,0 +1,105 @@
+name: publish distributions
+on:
+  push:
+    branches:
+    - main
+    tags:
+    - v*
+  pull_request:
+    branches:
+    - main
+    - release/v*
```

Comment:
This is a assuming a release branch pattern for future backports for bug fixes to patch versions of `release/v*` which may not be the case. This can be altered as desired.

Reply:
I don't *think* we need to worry about backports here. I'm trying to think of a situation where we'd want to do that. 

We generally try to avoid making backwards incompatible changes to the standard itself, but even if there was one, I think we would want to support multiple versions in the same array-api-compat version using [`api_version`](https://github.com/data-apis/array-api-compat/blob/ea6a9d695d011ff84c0b1ebd1cb624875fe9d131/array_api_compat/common/_helpers.py#L56). Similarly, we'll probably want to support both NumPy 2.0 and NumPy 1.25 at the same time when that happens.

Reply:
> I don't _think_ we need to worry about backports here.

Sounds good. I'll drop this then if you're not planning on making releases from any branch other than `main`.

Reply:
This has now been removed.